### PR TITLE
Block production should be able to unlock pending blocks

### PR DESCRIPTION
**Pull request by nearmax**
If we produce block X but there are blocks X+1, ... X+Y that we have already received through the announcements, then we should be able to progress to block X+Y.

### CODE REVIEW

File: node/client/src/lib.rs [hunk 1]
```diff
@@ -7,36 +7,37 @@ extern crate parking_lot;
...
 pub mod test_utils;
 
 const POISONED_LOCK_ERR: &str = "The lock was poisoned.";
-const BEACON_SHARD_BLOCK_MATCH: &str = "Expected to have shard block present when processing beacon block";
+const BEACON_SHARD_BLOCK_MATCH: &str =
+    "Expected to have shard block present when processing beacon block";
 
 /// Result of client trying to produce a block from a given consensus.
-#[allow(clippy::large_enum_variant)]  // This enum is no different from `Option`.
+#[allow(clippy::large_enum_variant)] // This enum is no different from `Option`.
```
Comment (bowenwang1996) [inline node/client/src/lib.rs]: We can use `Box` to avoid that

File: node/client/src/lib.rs [hunk 2]
```diff
@@ -349,15 +365,26 @@ impl Client {
...
-            .insert(shard_block);
+        self.pending_shard_blocks.write().expect(POISONED_LOCK_ERR).insert(shard_block);
         self.pending_beacon_blocks.write().expect(POISONED_LOCK_ERR).insert(beacon_block);
-        let best_block_hash = self.beacon_chain.chain.best_hash();
+        let best_block_hash = self.beacon_client.chain.best_hash();
 
+        let new_best_block = self.beacon_client.chain.best_block();
+        self.try_apply_pending_blocks();
+
+        if new_best_block.block_hash() == best_block_hash {
```
Comment (bowenwang1996) [inline node/client/src/lib.rs]: It seems that this equality always holds. Should `try_apply_pending_blocks` be called before getting `new_best_block`?

Comment (nearmax) [reply node/client/src/lib.rs]: Good catch. I was wondering why the tests are breaking..

File: node/client/src/lib.rs (hunk 1 above)
Comment (nearmax) [reply node/client/src/lib.rs]: Done.

**Comment by nearmax** [comment]
@bowenwang1996 PTAL

**Pull request merged**